src/cart/crt_hlc.c

```diff
@@ -82,6 +83,13 @@ uint64_t crt_hlc_get_msg(uint64_t msg)
 	uint64_t pt = crt_hlc_localtime_get();
 	uint64_t hlc, ret, ml = msg & ~CRT_HLC_MASK;
 
+	if (pt + crt_hlc_threshold < ml) {
+		/* remote node is far in future */
+		D_DEBUG(DB_ALL, "HLC is out of sync (%lu + %lu  < %lu)\n",
+			pt, crt_hlc_threshold, ml);
+		return ~0ULL;
```

Comment:
this happens at two cases,
1) server get the request msg from client, it finds client's HLC is far ahead, return -DER_TIME_SYNC seems fine.
2) client gets the reply msg from server, it finds server's HLC is far ahead (client is far behind), in this case besides return -DER_TIME_SYNC, can also sync client's local HLC as below code?

BTW, crt_hlc_threshold probably can set as NTP config's max clock offset (250mS?), but 2S maybe safer for now.